libs/ledgerjs/packages/hw-app-exchange/src/SwapUtils.ts

```diff
@@ -0,0 +1,59 @@
+import { loadSync } from "protobufjs";
+
+export type SwapProtobufPayload = {
+  payinAddress: string;
+  payinExtraId?: string;
+  refundAddress: string;
+  refundExtraId?: string;
+  payoutAddress: string;
+  payoutExtraId?: string;
+  currencyFrom: string;
+  currencyTo: string;
+  amountToProvider: Buffer;
+  amountToWallet: Buffer;
+  message?: string;
+  deviceTransactionId?: string;
+  deviceTransactionIdNg?: Buffer;
+};
+
+export type SwapPayload = {
+  payinAddress: string;
+  payinExtraId?: string;
+  refundAddress: string;
+  refundExtraId?: string;
+  payoutAddress: string;
+  payoutExtraId?: string;
+  currencyFrom: string;
+  currencyTo: string;
+  amountToProvider: number;
```

Comment:
I think we might have an issue with `number` type as it may be not large enough. We might consider to use `bigint` instead.

Reply:
Fixed